hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -686,9 +686,24 @@ def _validate_deadline(x):
                 "deadline=%r is invalid, because it is impossible to meet a "
                 "deadline <= 0.  Use deadline=None to disable deadlines." % (x,)
             )
+        # check if x is larger than timedelta's max value (1000000000 days - 1 microsecond)
+        if x >= 86400000000000000:
+            raise InvalidArgument(
+                "deadline=%r is invalid, because it is impossible to meet a "
+                "deadline > datetime.timedelta.max. Use deadline=None to disable deadlines." % (x,)
+            )
+        if x is None:
+            return x
+        return datetime.timedelta(microseconds=x * 1000)
+    if isinstance(x, datetime.timedelta):
+        if x.days < 0 or x == datetime.timedelta():
```

Comment:
For clarity, this can be a single condition:

```python
if x <= datetime.timedelta(0):
```

Having this check also means that we can delete the now-redundant version that checks the lower bound of the number above.